### DISCUSSION TITLE
Accessing additional details through DirectLine

### DISCUSSION THREAD

**Discussion by rschuetzler** [microsoft/BotFramework-Composer#9437] [Q&A]
I am building a bot with Bot Framework Composer, and I'm wondering if there's a way to get additional information to come through the DirectLine API. A couple things I would like:

1. I have variables that I set for each conversation that track the status of the conversation. I'd like to add some additional elements to my interface to reflect that status. Is there a way to pass variables such as `user.somevalue` through to my frontend?
2. Is there a way for me to get access to classification details, such as confidence (`turn.recognized.score`?)

Thanks!

**Comment by fausttiger007** [comment]
you can store your values via    Set Property(ties)

And retrieve them later as ${user.something}  , etc.

![image](https://user-images.githubusercontent.com/905135/212717211-cd00b5a5-c5b2-4dee-b89d-a6ace61656a0.png)

Not sure what API you're calling for your classification, but for custom question answering, the scores are in the returned API response.

If you setup Application Insights/Analytics Logs, you can also get the Scores from Power BI by querying the Language resource.

 ================

I also want to look into writing custom outputs from Bot Composer to the analytic logs for later reporting, 

- Debugging Options / Emit a telemetry track event ?
- Access external resources/Emit a custom event ?

...although not sure what tables these are written to (i.e. AppEvents  formerly Custom Events? etc...)    Some of the current QnAmaker events in Azure Bot logs (i.e. ActivityID, etc.) don't appear in the standard table/field definitions so I'm wondering if they are hard built into Bot Composer as QnAmaker specific outputs, so need to attempt to create my own with using Custom QA or other APIs.

Anyone here have input/experience here for external API workflows like Custom QuestionAnwering (not QnAmaker) ?

![image](https://user-images.githubusercontent.com/905135/212718420-8dbeb07c-9457-4c87-acd0-32d68624d16a.png)

OR
 
![image](https://user-images.githubusercontent.com/905135/212719927-41d353e1-8de0-4015-a62b-2f09ce8545aa.png)
